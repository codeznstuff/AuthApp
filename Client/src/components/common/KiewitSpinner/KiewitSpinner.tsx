import React from 'react';
import styles from './KiewitSpinner.module.scss';
import Spinner from 'react-bootstrap/Spinner';

interface SpinnerProps {
  animation?: 'border' | 'grow';
  role?: string;
  show?: boolean;
}

/* You can either toggle the "show" prop or just return the component without it */
export const KiewitSpinner = ({ animation = 'border', role = 'status', show = true, ...props }: SpinnerProps) => {
  if (!show) return null;

  return (
    <div className={styles.spinnerContainer}>
      <Spinner animation={animation} role={role} {...props} />
    </div>
  );
};
