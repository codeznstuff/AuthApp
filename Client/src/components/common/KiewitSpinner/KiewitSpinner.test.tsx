import React from 'react';
import { render } from '@testing-library/react';
import { KiewitSpinner } from './KiewitSpinner';

describe('KiewitSpinner', () => {
  it('does not render when show prop is false', () => {
    const spinner = render(<KiewitSpinner show={false} />);
    expect(spinner.container.firstChild).toBeNull();
  });

  it('renders when show is true or not passsed in', () => {
    const spinner = render(<KiewitSpinner />);
    expect(spinner.container.firstChild).not.toBeNull();
    expect(spinner.container.firstChild).toMatchSnapshot();
  });
});
