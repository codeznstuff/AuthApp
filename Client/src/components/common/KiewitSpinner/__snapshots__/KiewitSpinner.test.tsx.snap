// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KiewitSpinner renders when show is true or not passsed in 1`] = `
<div
  class="spinnerContainer"
>
  <div
    class="spinner-border"
    role="status"
  />
</div>
`;
